Feature/dapp quick pay route

weilbith opened the pull request:
**Thank you for submitting this pull request :)**

Fixes #2882

**Short description**
The most exciting feature for the dApp since the connection manager 😬 
All the previous refactoring enabled to do such nice thing like this.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [x] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Please checkout the image with the different cases in the linked issue.

weilbith commented:
I need to add a CHANGELOG entry.

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -275,7 +275,7 @@
       "too-many-decimals": "Please use only up to {decimals} decimals for this token",
       "not-enough-funds": "Your maximum usable amount is {funds} {symbol}",
-      "zero": "Enter an amount larger than zero",
+      "less-than-minimum": "Enter an amount larger than the minimum of {funds}",
```
"Enter an amount larger than {funds}"

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -731,6 +731,11 @@
+    },
+    "PFS_TARGET_NO_RECEIVE": {
+      "title": "Target does not receive payments",
+      "problem": "The target is online but does not accept to receive payments.",
```
"The target is online but does not seem to receive payments."

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -731,6 +731,11 @@
+    "PFS_TARGET_NO_RECEIVE": {
+      "title": "Target does not receive payments",
+      "problem": "The target is online but does not accept to receive payments.",
+      "solution": "Please contact the payment target to make sure he enables to receive payments and try the payment again."
```
"Please contact the payment target to make sure they can receive payments and try again."

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -304,6 +304,45 @@
+    "steps": {
+      "transfer": {
+        "title": "Transfering tokens",
+        "description": "Please wait for your payment"
```
"Please wait"

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -304,6 +304,45 @@
+    }
+  },
+  "channel-deposit-and-transfer-action": {
+    "confirm-button-label": "Deposit & Pay",
```
"Deposit and Pay"

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -304,6 +304,45 @@
+    }
+  },
+  "channel-open-and-transfer-action": {
+    "confirm-button-label": "Open Channel & Pay",
```
"confirm-button-label": "Open Channel and Pay"

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -875,5 +875,26 @@
+    "action-titles": {
+      "channel-deposit": "Channel Deposit",
+      "channel-open": "Open Channel",
+      "channel-deposit-and-transfer": "Channel Deposit & Pay",
```
"Channel Deposit and Pay"

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -875,5 +875,26 @@
+      "channel-deposit": "Channel Deposit",
+      "channel-open": "Open Channel",
+      "channel-deposit-and-transfer": "Channel Deposit & Pay",
+      "channel-open-and-transfer": "Channel Open & Pay"
```
"Channel Open and Pay"

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/locales/en.json:
```diff
@@ -875,5 +875,26 @@
+      "channel-deposit-and-transfer": "{deposit} will be deposited into the existing channel and {payment} will be paid out.",
+      "channel-open-and-transfer": "{deposit} will be deposited into a new channel and {payment} will be paid out."
+    },
+    "invalid-parameter-error": "Any of the given payment paramter is invalid. Please verify the used link is correct."
```
"Some payment parameters are invalid. Please verify that the link is correct."

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/views/QuickPayRoute.vue:
```diff
@@ -0,0 +1,354 @@
+  }
+
+  get actionDialogTitle(): string {
+    return this.actionComponent === TransferAction
```
Perhaps use `switch` here instead.

weilbith replied:
✔️

taleldayekh commented on raiden-dapp/src/views/QuickPayRoute.vue:
```diff
@@ -0,0 +1,354 @@
+  }
+
+  get actionMessagePath(): string {
+    return this.actionComponent === TransferAction
```
Perhaps use `switch` here instead.

weilbith replied:
✔️

taleldayekh reviewed (approved):
Went through this together with @weilbith and looks very good, thx! 👍🏼

weilbith commented:
@taleldayekh thanks for the nice review. I always appreciate that you take the time to do it together and discuss the details. Also thanks to always improve my user translations! 🙏🏾

weilbith commented:
I addressed all feedback. I'm very positive that I don't need an in-between CI confirmation. Therefore I'll go ahead and squash the fixups. Then I'll wait for CI to finally merge it.

weilbith commented:
Sorry, I messed something up... So much to I'm optimistic.
We really need some basic commit/push hooks.

andrevmatos commented:
@weilbith can you rebase on top of master on next run, please? some dependencies updates (including jest/test related stuff) got merged. Hopefully it won't impact actual code nor conflict, but it'll be better to have that tested before merging. Thanks

weilbith commented:
Rebase as requested by Andre...

weilbith commented:
Okay, apparently this did break something...

weilbith commented:
The Jest update changed the behavior of the `toContain` has changed.

weilbith commented:
I don't know why the `license/cla` check is not reporting. But I'm the only one who contributed code to this PR. And I have signed. So I use my permissions to merge this PR anyway.
